app/command_msg.go

```diff
@@ -58,6 +58,22 @@ func (me *msgProvider) DoCommand(a *App, args *model.CommandArgs, message string
 		return &model.CommandResponse{Text: args.T("api.command_msg.missing.app_error"), ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL}
 	}
 
+	user, err := a.Srv.Store.User().Get(args.UserId)
+	if err != nil {
+		mlog.Error(err.Error())
+		return &model.CommandResponse{Text: args.T("api.command_msg.fail.app_error"), ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL}
+	}
+	if user.IsGuest() {
```

Comment:
This is not about if the user is a guest or not, is about the visibility of the other user. I could (in the future) remove the permission to view other people, and that is not related to guest itself.

Reply:
I see. I think I was a bit misled by the fact that this behaviour is expected for normal users but not for guests. I guess `UserCanSeeOtherUser` implicitly checks who a guest can/cannot see.